### PR TITLE
[Gmail OAuth] Add update signature for org email

### PR DESCRIPTION
**Pull request by andrewjschuang**
Pending:

- [ ] Unpublish `gmail_custom_oauth-update-signature` after this PR is merged

Separates into two actions:

- Update Signature for Email in Organization
- Update Signature for Primary Email Address

For **Update Signature for Email in Organization**, the Google Cloud authentication is required.

See #4801.

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Comment by pd-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

File: components/gmail_custom_oauth/actions/update-org-signature/update-org-signature.mjs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import base from "../update-primary-signature/update-primary-signature.mjs";
+
+const docLink = "https://developers.google.com/gmail/api/reference/rest/v1/users.settings.sendAs/update";
+
+export default {
+  ...base,
+  key: "gmail_custom_oauth-update-org-signature",
+  name: "Update Signature for Email in Organization",
+  description: `Update the signature for a specific email address in an organization.
+    A Google Cloud service account with delegated domain-wide authority is required for this action. [See the docs.](${docLink})`,
```
Comment (lcaresia) [inline components/gmail_custom_oauth/actions/update-org-signature/update-org-signature.mjs:11]:
```suggestion

export default {
  ...base,
  key: "gmail_custom_oauth-update-org-signature",
  name: "Update Signature for Email in Organization",
  description: "Update the signature for a specific email address in an organization.
    A Google Cloud service account with delegated domain-wide authority is required for this action. [See docs here](https://developers.google.com/gmail/api/reference/rest/v1/users.settings.sendAs/update)",
```

File: components/gmail_custom_oauth/actions/update-primary-signature/update-primary-signature.mjs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import gmail from "../../gmail_custom_oauth.app.mjs";
+
+const docLink = "https://developers.google.com/gmail/api/reference/rest/v1/users.settings.sendAs/update";
+
+export default {
+  key: "gmail_custom_oauth-update-primary-signature",
+  name: "Update Signature for Primary Email Address",
+  description: `Update the signature for the primary email address. [See the docs.](${docLink})`,
```
Comment (lcaresia) [inline components/gmail_custom_oauth/actions/update-primary-signature/update-primary-signature.mjs:8]:
```suggestion

export default {
  key: "gmail_custom_oauth-update-primary-signature",
  name: "Update Signature for Primary Email Address",
  description: "Update the signature for the primary email address. [See docs here](https://developers.google.com/gmail/api/reference/rest/v1/users.settings.sendAs/update)",
```

**Review by lcaresia — DISMISSED** [review]
LGTM! Just some fixes!

**Review by lcaresia — APPROVED** [review]

**Pull request merged**